account_invoice_discount_formula/models/account_invoice.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2018 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError, Warning
```

Comment:
Avoid `Warning`, use `UserError` instead. This because Warning is a key work of python.